src/phoenix/experimental/evals/templates/template.py

```diff
@@ -15,19 +14,27 @@
 NOT_PARSABLE = "NOT_PARSABLE"
 
 
-@dataclass
 class PromptTemplate:
     text: str
     variables: List[str]
 
-    def __init__(self, text: str, delimiters: List[str] = [DEFAULT_START_DELIM, DEFAULT_END_DELIM]):
+    def __init__(
+        self,
+        text: str,
+        delimiters: List[str] = [DEFAULT_START_DELIM, DEFAULT_END_DELIM],
+    ):
         self.text = text
         self._start_delim, self._end_delim = self._get_delimiters(delimiters)
-        self._validate()
-        self.variables = self._parse_variables()
+        self._validate(self.text)
```

Comment:
clarification: this means the validate function is static and doesn't need to be on the class anymore and doesn't need to be attached to the class in this way? Just trying to understand the change in signature since now the function looks pure (which I like - I just don't know if needs to be on the class at this point)

Reply:
For the time being I was keeping all of the interfaces static to ensure this is close to a pure refactor. I'm actually not sure we need the `_validate` function? The regex function will ensure we only capture enclosed variables and I don't see why we need to prevent any stray delimiter characters existing in the template